src/charm_state.py

```diff
@@ -119,10 +124,16 @@ def from_charm(cls, charm: CharmBase) -> "CharmState":
         workers = charm.config.get("webserver_workers")
         threads = charm.config.get("webserver_threads")
         flask_config = {
-            k.removeprefix("flask_"): v for k, v in charm.config.items() if k.startswith("flask_")
+            k.removeprefix("flask_"): v
+            for k, v in charm.config.items()
+            if k.startswith("flask_") and k in KNOWN_CHARM_CONFIG
+        }
+        app_config = {
+            k.removeprefix("flask_"): v
+            for k, v in charm.config.items()
+            if k.startswith("flask_") and k not in KNOWN_CHARM_CONFIG
```

Comment:
I think there's a problem here:
1) I don't think we should remove any prefix from the app configuration, if the user used "flask_" they'll endup with an envvar "FLASK_FLASK_", we should keen them as they've been defined
2) Filtering out "known_charm_config" is usefull there, but doesn't guarantee potential envvar clash. Let's say a user defined a configuration named "application_root", this will be translated to "FLASK_APPLICATION_ROOT" which gets in conflict with a known_charm_config. Therefore a further filtering is required in the app_config

Reply:
1. Should we strip the `flask_` prefix for other builtin charm configuration, like `flask_ application_root` as well? For the sake of consistency?
2. The configuration conflict and merging is handled in the `FlaskApp.flask_environment` function.

Reply:
For 1) I'm not sure I get your point, you're already removing the prefix in the flask_config params.
What I want to highlight here is that we don't expect app_config to start with "flask_" and we don't modify their name (so don't remove any potential prefix).
So if a user set a config:
`my_app_salt` that's accepted and translated to `FLASK_MY_APP_SALT` envvar
If a user set a config:
`flask_admin_email` that's accepted and translated to `FLASK_FLASK_ADMIN_EMAIL`

The builtin config are stripped to have a consistent environment transformation (add prefix + uppercase)

For 2) Indeed I saw it while doing the review

Reply:
I mean the configuration names `flask_application_root` defined in the `config.yaml` file, should we remove the `flask_` from the built-in configuration names defined in the `config.yaml` file as well?

Reply:
I don't think so, the prefix in these configuration are there because they are specific to flask
The prefix for the envvar are there by convention for Flask to know which envvars to use

Even if the 2 prefixes are the same, they serve different purpose